test/extended/util/test_setup.go

```diff
@@ -71,9 +78,18 @@ func InitTest(dryRun bool) error {
 		}
 		TestContext.Host = cfg.Host
 	}
-
 	// Ensure that Kube tests run privileged (like they do upstream)
-	TestContext.CreateTestingNS = createTestingNS
+	TestContext.CreateTestingNS = func(ctx context.Context, baseName string, c kclientset.Interface, labels map[string]string) (*corev1.Namespace, error) {
+		// there are several cases when we want to kick in with namespace customization
+		// which includes setting persmissions for upstream tests to allow for privileged
+		// operations, these cases are:
+		// 1. all tests which namespace matches `e2e-k8s-...-upgrade` pattern
+		//    (mostly k8s upgrades tests, see test/extended/util/disruption/disruption.go#createTestFrameworks)
+		// 2. all k8s tests (based on testfile location), which don't have specific wording in their name (see skipTestNamespaceCustomization)
+		isKubeNamespace := upgradeFilter.MatchString(baseName) || // 1.
+			(isGoModulePath(ginkgo.CurrentSpecReport().FileName(), "k8s.io/kubernetes", "test/e2e") && !skipTestNamespaceCustomization()) // 2.
+		return e2e.CreateTestingNS(ctx, baseName, c, labels, isKubeNamespace)
+	}
```

Comment:
The behavior is roughly similar, the `e2e.CreateTestingNS` is in o/k because we also need explicitly set the namespaces as privileged for all k8s tests. For origin tests we've identified two cases described in this file:
1. when we run the embedded k8s tests (the path check) 
2. when we run the upgrade tests (those will be always embedded, based on my recent findings).

I'm planning to explicitly call it out these modifications for namespaces in https://github.com/openshift/enhancements/pull/1291 to ensure this is more widely known that it is today.